src/coreclr/vm/wasm/calldescrworkerwasm.cpp

```diff
@@ -13,7 +13,7 @@ extern "C" void STDCALL CallDescrWorkerInternal(CallDescrData * pCallDescrData)
     if (targetIp == NULL)
     {
         GCX_PREEMP();
-        pMethod->PrepareInitialCode(CallerGCMode::Coop);
+        (void)pMethod->DoPrestub(NULL /* MethodTable */, CallerGCMode::Preemptive);
```

Comment:
```suggestion
        (void)pMethod->DoPrestub(NULL /* MethodTable */, CallerGCMode::Coop);
```
I think this should be `CallerGCMode::Coop` (or the default `CallerGCMode::Unknown` that takes a slower path).

This argument represents that GC mode that the method is expected to be called in. It does not represent the current GC mode. `CallDescrWorkerInternal` is only ever used to invoke managed methods that are expected to be called in cooperative mode.

Reply:
Okay. I recall adding this and the name is very misleading then. I must have misunderstood something when I added it. An enum called `CallerGCMode` seems rather clear that it is the current state of the caller.

Reply:
It is the state that the original caller is in before any transitions. `DoPrestub` has preemptive mode contract, so it would not make sense to pass in mode of the immediate `DoPrestub` caller.